### PR TITLE
Grip to json

### PR DESCRIPTION
**Pull request by tzmoz123**

### CODE REVIEW

File: convertGRIPtoJSON.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import json
+from collections import OrderedDict
+from bs4.element import NavigableString  # bs4.element.NavigableString
+from bs4 import BeautifulSoup
+
+
+def iter_values(value, returned_list: list) -> list:
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Bad function naming ( I had to ;) ) and bad argument naming (returned_list -> you don't return it)

File: convertGRIPtoJSON.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    No need to return the list, lists are by reference
+    """
+    if type(value) != NavigableString:
+        for iter_value in value:
+            iter_values(iter_value, returned_list)
+    else:
+        returned_list.append(value)
+
+
+def find_values(bs_xml: BeautifulSoup) -> dict:
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Bad argument naming (give it a full name)

File: convertGRIPtoJSON.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        # Check if there is any values
+        if step.value is not None:
+            returned_dic[step['name']] = []
+            for value in step.find_all('value'):
+                # Check if there is any iter values
+                iter_values(value, returned_dic[step['name']])
+    return returned_dic
+
+
+def clean_dict(dic: dict) -> dict:
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Bad function and argument naming (give it a full name)

File: convertGRIPtoJSON.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    """
+    Because of the file format, some enters(\n) are being considered as values
+    The function as well replace all spaces(' ') in the key names with underlines ('_')
+    """
+    new_dic = {}
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Bad naming (give it a full name)

File: convertGRIPtoJSON.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+def edit_json(json_file_name: str, modifiers: dict):
+    """
+    Change Values in the modifiers according to the given dictionary
+    Note: Change ONLY the given modifiers, the rest will STAY as is
+    """
+    with open(json_file_name, "r") as file_handler:
+        data = json.load(file_handler, object_pairs_hook=OrderedDict)
+    # Run-on all the modifiers functions according to their names
+    for m in modifiers:
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Bad naming (give it a full name)

File: convertGRIPtoJSON.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+def Blur(opened_json_file: OrderedDict):
+    """Change the Blur Values in the JSON data"""
+    # We have different system so we need to convert this value
+    return opened_json_file
+
+
+def HSV_Threshold(opened_json_file: OrderedDict):
+    """Change the HSV_Threshold Values in the JSON data"""
+    return opened_json_file
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: What is the purpose of those functions

File: convertGRIPtoJSON.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+
+# def Find_Contours(opened_json_file: OrderedDict): # Not supported in GRIP (yet I suppose)
+#    return opened_json_files
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Delete this

File: convertGRIPtoJSON.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+
+if __name__ == '__main__':
+    # Open GRIP file as a string
+    with open('f.grip', 'rb') as gf:
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Get the grip file name as a command line argument

File: convertGRIPtoJSON.py [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        xml_data = gf.read()
+    # Convert the xml string to BeautifulSoup object
+    xml = BeautifulSoup(xml_data)
+    # Get dict of all values and name
+    values = find_values(xml)
+    # Clean and formatting the dictionary
+    values = clean_dict(values)
+    # print(values)
+    json_file = 'vision-framework\\examples\\example_try.json'
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Get the grip file name as a command line argument

File: convertGRIPtoJSON.py [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+    """
+    new_dic = {}
+    for key, value in dic.items():
+        key = key.replace(' ', '_')
+        value = list(filter(lambda a: a != '\n', value))
+        new_dic[key] = value
+    return new_dic
+
+
+def edit_json(json_file_name: str, modifiers: dict):
```
Comment (barhenkro) [inline convertGRIPtoJSON.py]: Good naming !

**Review by barhenkro — CHANGES_REQUESTED** [review]
Generally gooks fine.
Check flake8 problems in circle ci.
Fix the namings and other things

File: convertGRIPtoJSON.py (hunk 6 above)
Comment (tzmoz123) [reply convertGRIPtoJSON.py]: using them the integration between the GRIP file and JSON file occur

**Pull request merged**